include/adc.h

```diff
@@ -26,121 +26,314 @@ extern "C" {
  * @{
  */
 
+/** @brief ADC channel gain factors. */
+enum adc_gain {
+	ADC_GAIN_1_6, /**< x 1/6. */
+	ADC_GAIN_1_5, /**< x 1/5. */
+	ADC_GAIN_1_4, /**< x 1/4. */
+	ADC_GAIN_1_3, /**< x 1/3. */
+	ADC_GAIN_1_2, /**< x 1/2. */
+	ADC_GAIN_2_3, /**< x 2/3. */
+	ADC_GAIN_1,   /**< x 1. */
+	ADC_GAIN_2,   /**< x 2. */
+	ADC_GAIN_3,   /**< x 3. */
+	ADC_GAIN_4,   /**< x 4. */
+	ADC_GAIN_8,   /**< x 8. */
+	ADC_GAIN_16,  /**< x 16. */
+	ADC_GAIN_32,  /**< x 32. */
+	ADC_GAIN_64,  /**< x 64. */
+};
+
+/** @brief ADC references. */
+enum adc_reference {
+	ADC_REF_VDD_1,     /**< VDD. */
+	ADC_REF_VDD_1_2,   /**< VDD/2. */
+	ADC_REF_VDD_1_3,   /**< VDD/3. */
+	ADC_REF_VDD_1_4,   /**< VDD/4. */
+	ADC_REF_INTERNAL,  /**< Internal. */
+	ADC_REF_EXTERNAL0, /**< External, input 0. */
+	ADC_REF_EXTERNAL1, /**< External, input 1. */
+};
+
+/** Acquisition time is expressed in microseconds. */
+#define ADC_ACQ_TIME_MICROSECONDS  (1u)
+/** Acquisition time is expressed in nanoseconds. */
+#define ADC_ACQ_TIME_NANOSECONDS   (2u)
+/** Acquisition time is expressed in ADC ticks. */
+#define ADC_ACQ_TIME_TICKS         (3u)
+/** Macro for composing the acquisition time value in given units. */
+#define ADC_ACQ_TIME(unit, value)  (((unit) << 14) | ((value) & BIT_MASK(14)))
+/** Value indicating that the default acquisition time should be used. */
+#define ADC_ACQ_TIME_DEFAULT       0
+
+#define ADC_ACQ_TIME_UNIT(time)    (((time) >> 14) & BIT_MASK(2))
+#define ADC_ACQ_TIME_VALUE(time)   ((time) & BIT_MASK(14))
+
 /**
- * @brief ADC driver Sequence entry
- *
- * This structure defines a sequence entry used
- * to define a sample from a specific channel.
+ * @brief Structure for specifying the configuration of an ADC channel.
  */
-struct adc_seq_entry {
-	/** Clock ticks delay before sampling the ADC. */
-	s32_t sampling_delay;
```

Comment:
The API is intended to be generic, so it cannot directly expose all features of particular hardware. If I correctly understand the purpose of this sampling delay feature, you can use it to realize the intervals between samplings that can by specified in [sequence options](https://github.com/zephyrproject-rtos/zephyr/pull/7691/files#diff-895d709eadff47267669f2993f5ff80eR157), instead of involving a timer for this.

Reply:
Ok this makes sense.